app/geant-exporter/geant-exporter.cc

```diff
@@ -16,109 +16,37 @@
-    /*!
-     * Fetching the GDML input file and initializing the geometry
-     */
-    std::string gdml_input = argv[argc - 1];
-
-    run_manager->SetUserInitialization(new DetectorConstruction(gdml_input));
-
-    //-----------------------------------------------------------------------//
-    /*!
-     * Loading the physics list
-     */
-    std::vector<G4String>* physics_constructor = new std::vector<G4String>;
-    physics_constructor->push_back("G4EmStandardPhysics");
-
-    G4VModularPhysicsList* physics_list;
-
-    // Adding the user-defined physics list
-    physics_list = new G4GenericPhysicsList(physics_constructor);
-
-    // For the full Physics List:
-    // Uncomment #include "FTFP_BERT.hh" and use the following pointer instead
-    // physicsList = new FTFP_BERT;
-
-    // Initializing the Physics List
-    run_manager->SetUserInitialization(physics_list);
+#include <TFile.h>
+#include <TDirectory.h>
+#include <TTree.h>
+#include <TBranch.h>
 
-    //-----------------------------------------------------------------------//
-    /*!
-     * The ActionInitialization class is responsible for calling all the
-     * remaining geant classes
-     */
-    run_manager->SetUserInitialization(new ActionInitialization());
-
-    //-----------------------------------------------------------------------//
-    /*!
-     * Starting a minimal run to produce the physics tables
-     */
-    auto ui_manager = G4UImanager::GetUIpointer();
-    ui_manager->ApplyCommand("/run/initialize");
-
-    run_manager->BeamOn(1);
-
-    //-----------------------------------------------------------------------//
-    /*!
-     * Storing both particle definition information and physics tables
-     *
-     * Code was splitted into two separate loops for simplicity:
-     * - First loop stores particle info; second stores physics tables
-     * - Minimal runtime difference penalty in favor of an easier readability
-     */
-    //-----------------------------------------------------------------------//
-
-    auto particle_iterator = G4ParticleTable::GetParticleTable()->GetIterator();
-
-    //-----------------------------------------------------------------------//
-    /*!
-     * LOOP 1: Storing particle information
-     */
-    //-----------------------------------------------------------------------//
+// Project
+#include "DetectorConstruction.hh"
+#include "ActionInitialization.hh"
+#include "GeantPhysicsTableParser.hh"
 
-    // Creating the particle definition root file
-    G4String               root_output_filename = "geant-exporter-data.root";
+//---------------------------------------------------------------------------//
+/*!
+ * Loops over particles and stores their data
+ */
+void store_particles(std::string                         root_output_filename,
+                     G4ParticleTable::G4PTblDicIterator* particle_iterator)
+{
     std::unique_ptr<TFile> root_output(
         new TFile(root_output_filename.c_str(), "recreate"));
```

Comment:
Should probably update this to use Philippe's suggestion of `TFile::Open`.

Reply:
Done.